store/dbInstance.go

```diff
@@ -55,94 +56,99 @@ type DBInstance struct {
+			DatabaseEngine: offer.InstancePayload.DatabaseEngine,
+			Type:           offer.ChargeType,
+			Payload:        termPayload,
+			HourlyUSD:      offer.HourlyUSD,
+			CommitmentUSD:  offer.CommitmentUSD,
 		}
-		if termList, ok := offerMap[offer.InstanceID]; !ok {
-			offerMap[offer.InstanceID] = []*Term{term}
+		if _, ok := termMap[offer.ID]; !ok {
+			termMap[offer.ID] = []*Term{term}
 		} else {
-			offerMap[offer.InstanceID] = append(termList, term)
+			termMap[offer.ID] = append(termMap[offer.ID], term)
 		}
 	}
 
 	now := time.Now().UTC()
 	incrID := 0
-	for _, instance := range instanceList {
+	// dbInstanceMap is used to aggregate the instance by their type (e.g. db.m3.large).
+	dbInstanceMap := make(map[string]*DBInstance)
+	var dbInstanceList []*DBInstance
+	// extract dbInstance from the payload field stored in the offer.
+	for _, offer := range offerList {
+		// filter the offer does not have a instancePayload (only got price but no goods).
+		if offer.InstancePayload == nil {
+			continue
+		}
+
+		instance := offer.InstancePayload
 		vCPUInt, err := strconv.Atoi(instance.VCPU)
 		if err != nil {
 			return nil, fmt.Errorf("Fail to parse the VCPU value from string to int, [val]: %v", instance.VCPU)
 		}
-
 		memoryDigit := instance.Memory[:strings.Index(instance.Memory, "GiB")-1]
 
-		if dbInstance, ok := dbInstanceMap[instance.Type]; ok {
-			regionList := dbInstance.RegionList
-			isRegionExist := false
-			for _, region := range regionList {
-				if region.Name == instance.Region {
-					if offerList, ok := offerMap[instance.ID]; ok {
-						for _, offer := range offerList {
-							offer.DatabaseEngine = instance.DatabaseEngine
-						}
-						region.TermList = append(region.TermList, offerMap[instance.ID]...)
-					}
-					isRegionExist = true
-					break
-				}
-			}
-			if !isRegionExist {
-				regionList = append(regionList, &Region{
-					Name:     instance.Region,
-					TermList: offerMap[instance.ID],
-				})
-			}
-		} else {
+		// we use the instance type (e.g. db.m3.xlarge) differentiate the specification of each instances,
+		// and consider they as the same instance.
+		if _, ok := dbInstanceMap[instance.Type]; !ok {
 			dbInstance := &DBInstance{
-				ID:         incrID,
-				ExternalID: instance.ID,
-				RowStatus:  RowStatusNormal,
-				CreatorID:  SYSTEM_BOT,
-				CreatedTs:  now.Unix(),
-				UpdaterID:  SYSTEM_BOT,
-				UpdatedTs:  now.Unix(),
-				RegionList: []*Region{
-					{
-						Name:     instance.Region,
-						TermList: offerMap[instance.ID],
-					},
-				},
+				ID:            incrID,
+				RowStatus:     RowStatusNormal,
+				CreatorID:     SYSTEM_BOT,
+				CreatedTs:     now.Unix(),
+				UpdaterID:     SYSTEM_BOT,
+				UpdatedTs:     now.Unix(),
 				CloudProvider: CloudProviderAWS,
 				Name:          instance.Type, // e.g. db.t4g.xlarge
 				VCPU:          vCPUInt,
 				Memory:        memoryDigit,
 				Processor:     instance.PhysicalProcessor,
+
+				RegionList: []*Region{},
```

Comment:
We don't need to set "RegionList: []*Region{}," because that's the default for Golang.